### PR TITLE
Refactor placement config and client out of internal

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Refactor placement config and client out of internal package to allow control plane service to also connect to placement service.

No behavior change.

## Issue reference

Please reference the issue this PR will close: N/A

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/actors/placement/placement_interface.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package internal
```
Comment (ItalyPaleAle) [inline pkg/actors/placement/placement_interface.go:14]: Please create a new package for the interfaces, don't move them inside `placement` which is a concrete implementation.

File: pkg/actors/config/options.go [hunk 2]
```diff
@@ -11,7 +11,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package internal
```
Comment (ItalyPaleAle) [inline pkg/actors/config/options.go:14]: This file should not be in `config`. Maybe more in a package like `interfaces`, together with the placement interface.

**Review by ItalyPaleAle — CHANGES_REQUESTED** [review]

**Comment by artursouza** [comment]
Putting this on-hold for now. I might not need it.

File: pkg/actors/placement/placement_interface.go (hunk 1 above)
Comment (artursouza) [reply pkg/actors/placement/placement_interface.go:14]: No problem. Will move.

File: pkg/actors/config/options.go (hunk 2 above)
Comment (artursouza) [reply pkg/actors/config/options.go:14]: No problem. Will move.

**Comment by cicoyle** [comment]
If I remember correctly we were thinking we needed this for the Scheduler service if Scheduler knew about Placement, but we ended up not needing to go this route, so I'm closing this.

**Pull request closed without merging**